main.go

```diff
@@ -97,41 +103,81 @@ func main() {
+		failf("Error during upload: %s", err)
+	}
+}
+
+func parseFilesListConfig(fileList string) ([]releaseAsset, error) {
+	var assets []releaseAsset
+	if filelist := strings.TrimSpace(fileList); filelist != "" {
 		files := strings.Split(filelist, "\n")
-		for i, filePath := range files {
+		for _, filePath := range files {
 			if strings.TrimSpace(filePath) == "" {
 				continue
 			}
-
-			var fileName string
-			if s := strings.Split(filePath, "|"); len(s) > 1 {
-				if strings.TrimSpace(s[0]) != "" {
-					filePath = s[0]
-				} else {
-					failf("Invalid file path configuration: %s", filePath)
-				}
-				if strings.TrimSpace(s[1]) != "" {
-					fileName = s[1]
-				} else {
-					failf("Invalid file name configuration: %s", filePath)
-				}
-			} else {
-				fileName = filepath.Base(filePath)
-			}
-
-			log.Printf("(%d/%d) Uploading: %s - %s", i+1, len(files), fileName, filePath)
-			fi, err := os.Open(filePath)
+			fileName, filePath, err := getFileNameFromPath(filePath)
 			if err != nil {
-				failf("Failed to open file (%s), error: %s", filePath, err)
+				return nil, err
 			}
+			assets = append(assets, releaseAsset{filePath, fileName})
+		}
+	}
+	return assets, nil
+}
 
-			if _, _, err := client.Repositories.UploadReleaseAsset(context.Background(), owner, repo, newRelease.GetID(), &github.UploadOptions{Name: fileName}, fi); err != nil {
-				failf("Failed to upload file (%s), error: %s", filePath, err)
-			}
-			log.Donef("- Done")
+func uploadFileListWithRetry(assets []releaseAsset, client *github.Client, owner string, repo string, id int64) error {
+	fmt.Println()
+	log.Infof("Uploading assets:")
+	for i, asset := range assets {
+		log.Printf("(%d/%d) Uploading: %s - %s", i+1, len(assets), asset.displayFileName, asset.path)
+		fi, err := os.Open(asset.path)
+		if err != nil {
+			return fmt.Errorf("Failed to open file (%s), error: %s", asset.path, err)
+		}
+
+		if err := uploadFileWithRetry(asset.path, asset.displayFileName, fi, client, owner, repo, id); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func uploadFileWithRetry(filePath string, fileName string, fi *os.File, client *github.Client, owner string, repo string, id int64) error {
+	if err := retry.Times(3).Wait(5 * time.Second).Try(func(attempt uint) error {
+		if attempt > 0 {
+			log.Warnf("%d attempt failed", attempt)
+		}
+		if _, _, err := client.Repositories.UploadReleaseAsset(context.Background(), owner, repo, id, &github.UploadOptions{Name: fileName}, fi); err != nil {
+			return fmt.Errorf("Failed to upload file (%s), error: %s", filePath, err)
+		}
+		log.Donef("- Done")
+		return nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getFileNameFromPath(filePath string) (string, string, error) {
+	var fileName string
+	if s := strings.Split(filePath, "|"); len(s) > 1 {
+		if strings.TrimSpace(s[0]) != "" {
+			filePath = s[0]
+		} else {
+			return "", "", fmt.Errorf("Invalid file path configuration: %s", filePath)
```

Comment:
errors should begin with lowercase letter by Golang convention; check other occurences too.

Reply:
Fixed.